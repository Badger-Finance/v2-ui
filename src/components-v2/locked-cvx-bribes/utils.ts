import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
