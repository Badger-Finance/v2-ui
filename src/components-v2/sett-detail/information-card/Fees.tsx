import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import { StyledDivider } from '../styled';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
	specName: {
		fontSize: 12,
		lineHeight: '1.66',
	},
}));

export const Fees = (): JSX.Element => {
	const classes = useStyles();

	return (
		<Grid container>
			<Typography>Fees</Typography>
			<StyledDivider />
			<Grid container>
				<Grid container justify="space-between">
					<Typography className={classes.specName} color="textSecondary" display="inline">
						Withdrawal Fee
					</Typography>
					<Typography display="inline" variant="subtitle2">
						0.5%
					</Typography>
				</Grid>
				<Grid container justify="space-between">
					<Typography className={classes.specName} color="textSecondary" display="inline">
						Performance Fee
					</Typography>
					<Typography display="inline" variant="subtitle2">
						20.0%
					</Typography>
				</Grid>
			</Grid>
		</Grid>
	);
};
