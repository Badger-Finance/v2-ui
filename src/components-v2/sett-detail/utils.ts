import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
