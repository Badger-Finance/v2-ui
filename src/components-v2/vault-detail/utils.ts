import { Vault, VaultData } from '@badger-dao/sdk';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/vaults/vault-charts';
import { DelaySeverity } from '../../mobx/model/vaults/vault-rewards';

dayjs.extend(utc);

export const ChartModeTitles: Record<string, string> = {
	[ChartMode.Value]: 'Sett Value',
	[ChartMode.Ratio]: 'Token Ratio',
	[ChartMode.AccountBalance]: 'My Holdings',
	[ChartMode.BoostMultiplier]: 'Badger Boost',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};

export function defaultVaultBalance(vault: Vault): VaultData {
	return {
		address: vault.vaultToken,
		name: vault.name,
		symbol: vault.asset,
		pricePerFullShare: vault.pricePerFullShare,
		balance: 0,
		value: 0,
		earnedBalance: 0,
		earnedValue: 0,
		depositedBalance: 0,
		withdrawnBalance: 0,
		tokens: [],
		earnedTokens: [],
	};
}

export interface Balance {
	balance: number;
}

export function hasBalance(input?: Balance): boolean {
	if (!input) {
		return false;
	}
	return input.balance > 0;
}
