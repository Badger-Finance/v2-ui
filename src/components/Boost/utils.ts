import { makeStyles } from '@material-ui/core/styles';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import React from 'react';

type ComparisonConfig = {
  toCompareValue: number;
  toBeComparedValue: number;
  greaterCaseColor: string;
  lessCaseColor: string;
  defaultColor: string;
};

export const getColorFromComparison = ({
  toCompareValue,
  toBeComparedValue,
  greaterCaseColor,
  lessCaseColor,
  defaultColor,
}: ComparisonConfig): string => {
  if (toCompareValue > toBeComparedValue) return greaterCaseColor;
  if (toCompareValue < toBeComparedValue) return lessCaseColor;

  return defaultColor;
};

export const useAssetInputStyles = (
  currentValue: string,
  toCompare = 0,
): ((props?: React.CSSProperties) => ClassNameMap<'assetColor'>) => {
  return makeStyles((theme) => {
    const defaultColor = currentValue ? theme.palette.text.primary : theme.palette.text.secondary;
    const fontColor = getColorFromComparison({
      toCompareValue: Number(currentValue),
      toBeComparedValue: toCompare,
      greaterCaseColor: '#74D189',
      lessCaseColor: theme.palette.error.main,
      defaultColor,
    });

    return {
      assetColor: {
        color: fontColor,
      },
    };
  });
};
