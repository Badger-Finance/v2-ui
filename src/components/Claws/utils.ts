import BigNumber from 'bignumber.js';
import { INVALID_REASON } from './index';

export interface ValidationParams {
	amount: string | BigNumber;
	maximum: BigNumber;
	minimum?: BigNumber;
}

export function validateAmountBoundaries({
	amount,
	maximum,
	minimum = new BigNumber('0'),
}: ValidationParams): INVALID_REASON | undefined {
	let error: INVALID_REASON | undefined = undefined;

	const input = new BigNumber(amount);
	const amountExceedsBalance = input.gt(maximum);
	const amountIsLessThanMinimum = input.lt(minimum);

	if (amountIsLessThanMinimum) error = INVALID_REASON.UNDER_MINIMUM;
	if (amountExceedsBalance) error = INVALID_REASON.OVER_MAXIMUM;

	console.log({
		input: input.toString(),
		minimum: minimum.toString(),
		maximum: maximum.toString(),
		amountExceedsBalance,
		amountIsLessThanMinimum,
		error,
	});

	return error;
}
