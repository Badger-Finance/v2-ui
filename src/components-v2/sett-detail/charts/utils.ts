import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
