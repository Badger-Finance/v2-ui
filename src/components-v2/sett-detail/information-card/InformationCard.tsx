import React from 'react';
import { CardContainer } from '../styled';
import { Value } from './Value';
import { Tokens } from './Tokens';
import { Claims } from './Claims';
import { Harvests } from './Harvests';
import { Fees } from './Fees';

export const InformationCard = (): JSX.Element => {
	return (
		<CardContainer>
			<Value />
			<Tokens />
			<Claims />
			<Harvests />
			<Fees />
		</CardContainer>
	);
};
