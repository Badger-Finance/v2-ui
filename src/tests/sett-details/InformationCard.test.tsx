import React from 'react';
import { checkSnapshot } from 'tests/utils/snapshots';
import { InformationCard } from '../../components-v2/sett-detail/information-card/InformationCard';

describe('Information Card', () => {
	it('displays sett information', () => {
		checkSnapshot(<InformationCard />);
	});
});
