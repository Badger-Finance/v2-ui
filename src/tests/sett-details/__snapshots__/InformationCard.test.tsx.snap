// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information Card displays sett information 1`] = `
<div>
  <div
    class="MuiPaper-root-3 MuiCard-root-2 WithStyles(ForwardRef(Card))-root-1 MuiPaper-elevation1-7 MuiPaper-rounded-4"
  >
    <div
      class="MuiGrid-root-33 makeStyles-root-31 MuiGrid-container-34"
    >
      <p
        class="MuiTypography-root-136 MuiTypography-body1-138"
      >
        Value
      </p>
      <hr
        class="MuiDivider-root-168 WithStyles(ForwardRef(Divider))-root-166 WithStyles(ForwardRef(Divider))-root-167"
      />
      <p
        class="MuiTypography-root-136 makeStyles-amount-32 MuiTypography-body1-138"
      >
        $7,921,648.24
      </p>
    </div>
    <div
      class="MuiGrid-root-33 makeStyles-root-175 MuiGrid-container-34"
    >
      <p
        class="MuiTypography-root-136 MuiTypography-body1-138"
      >
        Tokens
      </p>
      <hr
        class="MuiDivider-root-168 WithStyles(ForwardRef(Divider))-root-166 WithStyles(ForwardRef(Divider))-root-181"
      />
      <div
        class="MuiGrid-root-33 MuiGrid-container-34"
      >
        <div
          class="MuiGrid-root-33 makeStyles-tokenSpec-177 MuiGrid-container-34 MuiGrid-justify-xs-space-between-53"
        >
          <div
            class="MuiBox-root-182 MuiBox-root-183"
          >
            <div
              class="makeStyles-tokenImageContainer-179"
            >
              <img
                alt="wbtc-icon"
                class="makeStyles-tokenImage-180"
                src="/assets/icons/wbtc.svg"
              />
            </div>
            <p
              class="MuiTypography-root-136 makeStyles-specName-176 makeStyles-tokenName-178 MuiTypography-body1-138 MuiTypography-colorTextSecondary-162 MuiTypography-displayInline-164"
            >
              WBTC
            </p>
          </div>
          <h6
            class="MuiTypography-root-136 MuiTypography-subtitle2-148 MuiTypography-displayInline-164"
          >
            114
          </h6>
        </div>
        <div
          class="MuiGrid-root-33 makeStyles-tokenSpec-177 MuiGrid-container-34 MuiGrid-justify-xs-space-between-53"
        >
          <div
            class="MuiBox-root-182 MuiBox-root-184"
          >
            <div
              class="makeStyles-tokenImageContainer-179"
            >
              <img
                alt="digg-icon"
                class="makeStyles-tokenImage-180"
                src="/assets/icons/digg.png"
              />
            </div>
            <p
              class="MuiTypography-root-136 makeStyles-specName-176 makeStyles-tokenName-178 MuiTypography-body1-138 MuiTypography-colorTextSecondary-162 MuiTypography-displayInline-164"
            >
              DIGG
            </p>
          </div>
          <h6
            class="MuiTypography-root-136 MuiTypography-subtitle2-148 MuiTypography-displayInline-164"
          >
            123
          </h6>
        </div>
        <div
          class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-justify-xs-space-between-53"
        >
          <p
            class="MuiTypography-root-136 makeStyles-specName-176 MuiTypography-body1-138 MuiTypography-colorTextSecondary-162 MuiTypography-displayInline-164"
          >
            Token Ratio
          </p>
          <h6
            class="MuiTypography-root-136 MuiTypography-subtitle2-148 MuiTypography-displayInline-164"
          >
            1.0789
          </h6>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-33 makeStyles-root-185 MuiGrid-container-34"
    >
      <p
        class="MuiTypography-root-136 MuiTypography-body1-138"
      >
        Claims
      </p>
      <hr
        class="MuiDivider-root-168 WithStyles(ForwardRef(Divider))-root-166 WithStyles(ForwardRef(Divider))-root-191"
      />
      <div
        class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-align-items-xs-center-42"
      >
        <div
          class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-item-35 MuiGrid-grid-xs-9-76"
        >
          <p
            class="MuiTypography-root-136 makeStyles-rewardsFrequency-186 MuiTypography-body1-138 MuiTypography-colorTextSecondary-162"
          >
            Reward Frequency
          </p>
          <span
            class="MuiTypography-root-136 makeStyles-frequencyDetail-187 MuiTypography-caption-139 MuiTypography-colorTextSecondary-162"
          >
            This Settâ€™s rewards are currently taking longer than usual.
          </span>
          <a
            class="MuiTypography-root-136 MuiLink-root-192 MuiLink-underlineHover-194 makeStyles-infoLink-188 MuiTypography-colorPrimary-159"
          >
            See more
          </a>
        </div>
        <div
          class="MuiGrid-root-33 makeStyles-rewardContainer-190 MuiGrid-item-35 MuiGrid-grid-xs-true-67"
        >
          <p
            class="MuiTypography-root-136 makeStyles-reward-189 MuiTypography-body1-138 MuiTypography-displayInline-164"
          >
            ~2 Hours
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-33 makeStyles-root-198 MuiGrid-container-34"
    >
      <p
        class="MuiTypography-root-136 MuiTypography-body1-138"
      >
        Harvests
      </p>
      <hr
        class="MuiDivider-root-168 WithStyles(ForwardRef(Divider))-root-166 WithStyles(ForwardRef(Divider))-root-203"
      />
      <div
        class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-align-items-xs-center-42"
      >
        <div
          class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-item-35 MuiGrid-grid-xs-9-76"
        >
          <p
            class="MuiTypography-root-136 makeStyles-harvestFrequency-199 MuiTypography-body1-138 MuiTypography-colorTextSecondary-162"
          >
            Harvest Frequency
          </p>
          <a
            class="MuiTypography-root-136 MuiLink-root-192 MuiLink-underlineHover-194 makeStyles-infoLink-200 MuiTypography-colorPrimary-159"
          >
            See Sett Harvest Stats
          </a>
        </div>
        <div
          class="MuiGrid-root-33 makeStyles-harvestContainer-202 MuiGrid-item-35 MuiGrid-grid-xs-true-67"
        >
          <p
            class="MuiTypography-root-136 makeStyles-harvest-201 MuiTypography-body1-138 MuiTypography-displayInline-164"
          >
            ~24 Hours
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-33 MuiGrid-container-34"
    >
      <p
        class="MuiTypography-root-136 MuiTypography-body1-138"
      >
        Fees
      </p>
      <hr
        class="MuiDivider-root-168 WithStyles(ForwardRef(Divider))-root-166 WithStyles(ForwardRef(Divider))-root-205"
      />
      <div
        class="MuiGrid-root-33 MuiGrid-container-34"
      >
        <div
          class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-justify-xs-space-between-53"
        >
          <p
            class="MuiTypography-root-136 makeStyles-specName-204 MuiTypography-body1-138 MuiTypography-colorTextSecondary-162 MuiTypography-displayInline-164"
          >
            Withdrawal Fee
          </p>
          <h6
            class="MuiTypography-root-136 MuiTypography-subtitle2-148 MuiTypography-displayInline-164"
          >
            0.5%
          </h6>
        </div>
        <div
          class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-justify-xs-space-between-53"
        >
          <p
            class="MuiTypography-root-136 makeStyles-specName-204 MuiTypography-body1-138 MuiTypography-colorTextSecondary-162 MuiTypography-displayInline-164"
          >
            Performance Fee
          </p>
          <h6
            class="MuiTypography-root-136 MuiTypography-subtitle2-148 MuiTypography-displayInline-164"
          >
            20.0%
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;
